src/contract.ts

```diff
@@ -134,11 +115,65 @@ export class AbstractContract {
     }
   }
 
+
+  getStateScript(states: Record<string, SupportedParamType>): Script {
```

Comment:
Add a detailed comment. This is an important function.

What happens when some stateful properties r missing?

```js
@state
int counter;

@state
bool flag;


// Only counter is passed
{
  counter: 12;
}
```